src/components/ModularTable/ModularTable.tsx

```diff
@@ -171,20 +171,54 @@ function ModularTable<D extends Record<string, unknown>>({
 
   const showEmpty: boolean = !!emptyMsg && (!rows || rows.length === 0);
 
+  // Function returns whether table can be sorted by a specific column.
+  // Returns true if sorting is enabled for the column and there is text
+  // being rendered within the column header, otherwise returns false.
+  const isColumnSortable = (column: HeaderGroup<D>) => {
+    if (!column.canSort) {
+      return false;
+    }
+
+    const headerNodesQueue: ReactNode[] = [column.render("Header")];
+    // Go through all header child nodes and search for text.
+    while (headerNodesQueue.length) {
+      const frontNode = headerNodesQueue.shift();
+      if (isValidElement(frontNode)) {
+        headerNodesQueue.push(frontNode.props.children);
+      }
+      if (
+        !isValidElement(frontNode) &&
+        (typeof frontNode === "string" || typeof frontNode === "number") &&
+        !!String(frontNode).trim()
+      ) {
+        return true;
+      }
+      if (!isValidElement(frontNode) && Array.isArray(frontNode)) {
+        frontNode.forEach((childNode) => headerNodesQueue.push(childNode));
+      }
+    }
+
+    return false;
```

Comment:
This works really nicely, but now that I see the complexity of this I wonder if we should only disable it if the user doesn't provide `Header` or provides a string of whitespace, but not disable it if they provide JSX that doesn't have any content (and if they want it disabled when they've provided empty JSX they need to pass `disableSortBy: true`)

If we did that then I think this could all be replaced with something like: `return typeof column.Header === "string" ? !!column.Header.trim() : isValidElement(column.Header)`.

This would also allow headers that have content provided by CSS e.g. icons such as `<i className="p-icon--smiley" />` (not sure if we actually do that anywhere but I can't see a reason to not allow it).